### PR TITLE
Docs for snapshots as simple archives

### PR DESCRIPTION
**Pull request by ywelsch**
Adds documentation for the new snapshots as archive feature.

Relates #81210

### CODE REVIEW

**Comment by debadair** [comment]
@elasticmachine run elasticsearch-ci/docs

**Review by javanna — APPROVED** [review]
left a couple of questions, LGTM otherwise

File: docs/reference/upgrade/archive-indices.asciidoc [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+Old mappings are imported as much "as-is" as possible into {es} 8, but only
+provide regular query capabilities on a select subset of fields:
+
+- <<number,Numeric types>>
+- <<boolean,`boolean` type>>
+- <<ip,`ip` type>>
+- <<geo-point,`geo_point` type>>
+- <<date,`date` types>>: the date `format` setting on date fields is supported
+  in so far as it behaves similarly across these versions. In case it is not,
```
[the remark this answers is not in the document]
Comment (ywelsch) [reply docs/reference/upgrade/archive-indices.asciidoc:31]: We have changed the meaning of various custom date formats / date literals across versions (e.g. year of era becomes week-based year), see https://www.elastic.co/guide/en/elasticsearch/reference/7.17/migrate-to-java-time.html#java-time-migration-incompatible-date-formats

Perhaps I could link to the above for additional context, as well as limit the scope of the problem to "custom date formats".

File: docs/reference/upgrade/archive-indices.asciidoc [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+- <<mapping-source-field,`_source` field>>
+
+{es} 5 indices with mappings that have {ref-7x}/removal-of-types.html[multiple mapping types]
+are collapsed together on a best-effort basis before they are imported.
+
+In case the auto-import of mappings does not work, or the new version
+can't make sense of the mapping, it falls back to a lightweight import of
+the mapping where the original mapping is stored in the _meta section of
+the imported index's mapping, and relies on the user to put the relevant
+mapping parts manually in place.
```
[the remark this answers is not in the document]
Comment (ywelsch) [reply docs/reference/upgrade/archive-indices.asciidoc:56]: I can clarify the process a bit more (e.g. user can inspect the original mapping using get mappings API, and use put mappings API to add selected portions back to index). Additionally I would like to ask users to report failures to auto-import mapping to us (but with the expectation that we will treat it as low-priority items to be fixed).

File: docs/reference/upgrade/archive-indices.asciidoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (javanna) [reply docs/reference/upgrade/archive-indices.asciidoc:31]: sounds good to expand a bit, thanks.

File: docs/reference/upgrade/archive-indices.asciidoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (javanna) [reply docs/reference/upgrade/archive-indices.asciidoc:56]: sounds great, I was also wondering if users know what you mean with "_meta" section. I would have to look it up myself :)

**Comment by ywelsch** [comment]
The release highlight changelog (https://github.com/elastic/elasticsearch/pull/86261/commits/cea246982e6db8dddacb8b1a10f835d94e2b7a06) seems to have been deleted by our automation. I will add it for the backport to 8.3 branch and mark that backport PR then as release highlight.

File: docs/reference/upgrade/archive-indices.asciidoc [hunk 3]
```diff
@@ -49,11 +50,16 @@ provide regular query capabilities on a select subset of fields:
...
-mapping parts manually in place.
+In case the auto-import of mappings does not work, or the new {es} version
+can't make sense of the mapping, it falls back to importing the index without
+the mapping, but stores the original mapping in the <<mapping-meta-field,_meta>>
+section of the imported index. The legacy mapping can then be introspected
+using the <<indices-get-mapping,GET mapping>> API and an updated mapping can
+be manually put in place using the <<indices-put-mapping,update mapping>> API,
+copying and adapting relevant sections of the legacy mapping to work with the
+current {es} version. While auto-import is expected to work in most cases,
+failures of doing so should be raised with the Elastic team for future
```
Comment (javanna) [inline docs/reference/upgrade/archive-indices.asciidoc:61]: link to open an issue?

File: docs/reference/upgrade/archive-indices.asciidoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (javanna) [reply docs/reference/upgrade/archive-indices.asciidoc:31]: do you mean "is supported as long as it behaves similarly across versions"?

File: docs/reference/upgrade/archive-indices.asciidoc [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+- <<boolean,`boolean` type>>
+- <<ip,`ip` type>>
+- <<geo-point,`geo_point` type>>
+- <<date,`date` types>>: the date `format` setting on date fields is supported
+  in so far as it behaves similarly across these versions. In case it is not,
+  for example {ref-7x}/migrate-to-java-time.html[when using custom date formats],
+  this field can be updated on legacy indices so that it can be changed by a
+  user if need be.
+- <<keyword-field-type,`keyword` type>>: the `normalizer` setting on keyword
+  fields is supported in so far as it behaves similarly across these versions.
```
Comment (javanna) [inline docs/reference/upgrade/archive-indices.asciidoc:36]: here too "is supported as long as..."

File: docs/reference/upgrade/archive-indices.asciidoc [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  In case it is not, this field can be updated on legacy indices if need be.
+- <<text-field-type,`text` type>>: scoring capabilities are limited, and all
+  queries return constant scores that are equal to 1.0. The `analyzer`
+  settings on text fields are supported in so far as they behave similarly
```
Comment (javanna) [inline docs/reference/upgrade/archive-indices.asciidoc:40]: supported as long as ?

**Comment by ywelsch** [comment]
Thanks @javanna!

**Pull request merged**